x-pack/plugins/apm/server/routes/fleet/api_keys/add_api_key_to_policies_if_missing.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cloneDeep, get, omit, set } from 'lodash';
+import { NewPackagePolicy, PackagePolicy } from '@kbn/fleet-plugin/common';
+import { CoreStart, Logger } from '@kbn/core/server';
+import { FleetStartContract } from '@kbn/fleet-plugin/server';
+import { getInternalSavedObjectsClient } from '../../../lib/helpers/get_internal_saved_objects_client';
+import { APMPluginStartDependencies } from '../../../types';
+import { getApmPackagePolicies } from '../get_apm_package_policies';
+import { createApmServerApiKey } from './create_apm_server_api_key';
+
+const API_KEY_PATH = `inputs[0].config['apm-server'].value.agent.config.elasticsearch.api_key`;
```

Comment:
**question**: can we change this to inject into the source_mapping section ? Is there any particular reason this path is being used ?

This was spotted while reviewing the related apm-server PR: https://github.com/elastic/apm-server/pull/9722#discussion_r1058954208

cc @simitt

Reply:
@axw suggested this path so it would match with @carsonip's change in https://github.com/elastic/apm-server/pull/9720

I can change it though. What do you suggest the exact path should be instead?

Reply:
Btw. the key is not only used for source maps. It's also used for retrieving agent configs.

Reply:
Can it be injected into the path for agent config (what is currently used) and the path for sourcemapping? The APM Server offers a dedicated configuration path for both features as they are independent of each other. The fact that we use one key for both with privileges for both should rather be an implementation detail on the Kibana side.

Reply:
That makes sense. I'll take a look tomorrow but yes, I don't think it's a problem injecting it twice.

Reply:
@simitt @kruskall I'm now injecting two different api keys: one with only access to the source map index, another with only access to agent config index.

Reply:
@kruskall Here's the full config as it looks now. I've just thrown things in without further consideration. Please let me know how the preferred structure should be. 

```json
          "config": {
            "apm-server": {
              "value": {
                "agent_config": [
                  {
                    "service": {},
                    "etag": "0d8423d58d01a04ed7eae94eed94632a8d152314",
                    "config": { "transaction_max_spans": "22" }
                  }
                ],
                "agent": {
                  "config": {
                    "elasticsearch": {
                      "api_key": "k22MhoUBlRqh1NK7iPYK:4ieFNXQeR1mWkBErzgHCCw}"
                    }
                  }
                },
                "rum": {
                  "source_mapping": {
                    "metadata": [
                      {
                        "service.name": "uploading-test",
                        "service.version": "1.0.0",
                        "sourcemap.url": "/api/fleet/artifacts/uploading-test-1.0.0/2f5d4e64ffde4acde832039186ca1652ed315fb0ecbcc1b398677b3bcba521df",
                        "bundle.filepath": "bar"
                      }
                    ],
                    "api_key": "lW2MhoUBlRqh1NK7ifbT:btwmjfVyTMaQKKSJzoc6XQ}"
                  }
                }
              }
            }
          },
```

Reply:
@sqren Thank you for this! :bow: 

Sorry I was a bit unclear in my comment. I meant under the `elasticsearch` section in `sourcre_mapping` (https://github.com/elastic/apm-server/blob/main/apm-server.yml#L192) according to https://github.com/elastic/apm-server/pull/9722#discussion_r1058954208

`rum.source_mapping.api_key` -> `rum.source_mapping.elasticsearch.api_key`

Reply:
No worries, thanks for the pointers. I've updated it to `rum.source_mapping.elasticsearch.api_key`.

Reply:
Thank you! :bow: